unit-tests/src/test/java/org/eclipse/collections/impl/block/function/Functions2Test.java

```diff
@@ -32,6 +37,54 @@ public void throwing()
                 }).value(null, null));
     }
 
+    @Test
+    public void throwingWithSuccessfulCompletion()
+    {
+        Assert.assertEquals("abcdef",
+                Functions2.throwing((ThrowingFunction2<String, String, String>)
```

Comment:
Is the cast necessary here?

Reply:
yes, when I remove the cast I get the compile failure:
Compilation failure
/unit-tests/src/test/java/org/eclipse/collections/impl/block/function/Functions2Test.java:[45,59] error: cannot find symbol
  symbol:   method concat(Object)
  location: variable argument1 of type Object

Reply:
Ok, got it. Then let's extract the lambda into a variable with that type. I realize this is accomplishing the essentially the same things but will make the code easier to read. Let's do that on the other instances as well. Thanks!

Reply:
Done!

Reply:
Can you move the ThrowingFunction inside into a separate variable?

Reply:
Done.